### PR TITLE
Support Volta+ arch in custatevec backend

### PR DESCRIPTION
**Pull request by 1tnguyen**
### Description

We fixed the `-gencode` option to `sm_80` (Ampere/A100) while the `CuStateVecCircuitSimulator.cu` does have some CUDA kernels (e.g.,  `__global__ void initializeDeviceStateVector`). Hence, the generated binary is not compatible with the V100 GPU.

In this case, the `initializeDeviceStateVector` was not executed since the runtime couldn't find a compatible binary to run; hence the state vector was left uninitialized (all 0.0's).

Fixed by modifying the `-gencode` to support more architectures: same level of compatibility as the cuquantum library.

Should fix the issues reported in https://github.com/NVIDIA/cuda-quantum/issues/623; https://github.com/NVIDIA/cuda-quantum/issues/622; https://github.com/NVIDIA/cuda-quantum/issues/583

### CODE REVIEW

**Review by amccaskey — APPROVED** [review]

File: .github/workflows/publishing.yml
Comment (1tnguyen) [inline .github/workflows/publishing.yml:1]: We need to trigger a `Basic validation (GPU backends)` run to verify the fix on the runner.

**Review by bettinaheim — APPROVED** [review]
Awesome, thanks for taking a look at these.

**Pull request merged**